### ISSUE TITLE
startWithMonth overwrites daily/weekly calendar's end date

### ISSUE THREAD

**Issue by Ane-tta** [kylestetz/CLNDR#234]
Hi there! Currently I'm developing a weekly calendar and when I want to set a custom date after using startWithMonth option, my weekly calendar starts showing all dates of month, but it has to show only the full week with custom date. Is there any way to achieve this?

**Comment by mikegioia** [comment]
Can you please post your configuration and template? I'd need to see what your working with to recommend anything.

**Comment by Ane-tta** [comment]
Part of my template:

<table class="table">
                <thead>
                    <tr>
                    <% _.each(daysOfTheWeek, function(day) { %>
                        <th class="<%= day %>"><%= day %></th>
                    <% }); %>
                    </tr>
                    <tr>
                    <% _.each(days, function(day) { %>
                        <th class="<%= day.classes %>"><%= day.day %> <%= intervalStart.format('MMM') %></th>
                    <% }); %>
                    </tr>
                </thead>
                <tbody>
                   ...
                </tbody>
<table class="table">

And configurations:
var calendar = $('#calendar').clndr({
    //startWithMonth:  moment(),
    startWithMonth: "2016-01-01",
    template: $('#full-clndr-template').html(),
    weekOffset: 1,
    daysOfTheWeek: ['SUN', 'MON', 'TUE', 'WED', 'THU', 'FRI', 'SAT'],
    lengthOfTime: {
      days: 7,
      interval: 7
    },
    targets: {
        nextButton: 'schedule-arrows-next',
        previousButton: 'schedule-arrows-prev'
    },
    events: lotsOfEvents,
    multiDayEvents: {
        endDate: 'end',
        singleDay: 'date',
        startDate: 'start'
    },
    clickEvents: {
      click: function(target) {
        console.log(target);
      },
    },
});

**Comment by Ane-tta** [comment]
Part of my template:
```
<table class="table">
                <thead>
                    <tr>
                    <% _.each(daysOfTheWeek, function(day) { %>
                        <th class="<%= day %>"><%= day %></th>
                    <% }); %>
                    </tr>
                    <tr>
                    <% _.each(days, function(day) { %>
                        <th class="<%= day.classes %>"><%= day.day %> <%= intervalStart.format('MMM') %></th>
                    <% }); %>
                    </tr>
                </thead>
                <tbody>
                   ...
                </tbody>
<table class="table">

And configurations:
var calendar = $('#calendar').clndr({
    //startWithMonth:  moment(),
    startWithMonth: "2016-01-01",
    template: $('#full-clndr-template').html(),
    weekOffset: 1,
    daysOfTheWeek: ['SUN', 'MON', 'TUE', 'WED', 'THU', 'FRI', 'SAT'],
    lengthOfTime: {
      days: 7,
      interval: 7
    },
    targets: {
        nextButton: 'schedule-arrows-next',
        previousButton: 'schedule-arrows-prev'
    },
    events: lotsOfEvents,
    multiDayEvents: {
        endDate: 'end',
        singleDay: 'date',
        startDate: 'start'
    },
    clickEvents: {
      click: function(target) {
        console.log(target);
      },
    },
});
```

**Comment by mikegioia** [comment]
@Ane-tta I think you found a bug. I haven't confirmed it yet, but my hunch is that the error is happening here: https://github.com/kylestetz/CLNDR/blob/master/src/clndr.js#L228

Line 228 of the library.:

```javascript
if (this.options.startWithMonth) {
    this.month = moment(this.options.startWithMonth).startOf('month');
    this.intervalStart = moment(this.month);
    this.intervalEnd = moment(this.month).endOf('month'); // <- BAD LINE
}
```

This is setting the ending interval to be the end of the month, even if the `lengthOfTime.days` options is specified (like in your case).

The workaround is to use `lengthOfTime.startTime` instead of `startWithMonth` until I get this fixed.

**Comment by mikegioia** [comment]
@Ane-tta Sorry for the long delay in this, but I pushed up a fix in #269. If you are able to, could you please pull that branch and test your app against it? I'm almost certain it's 100% working, as I tested it in the test file after re-creating your bug, but if you're able to confirm it I'll compile this directly to 1.4.4. Thanks again for reporting the bug.

**Comment by mikegioia** [comment]
This is up in 1.4.4. It works :P
